lib/utils/dns.go

```diff
@@ -169,8 +170,10 @@ func DNSReadConfig(rdr io.Reader) (*DNSConfig, error) {
 		return nil, err
 	}
 
-	if len(conf.Servers) == 0 {
-		conf.Servers = append(conf.Servers, defaultNS...)
+	if !noDefaultNameservers {
```

Comment:
I would invert the condition (`addDefaultNameservers`) - double negative hurts the brain.